Automattic/wp-calypso
# SSR The config module

Calypso currently has different configuration for its different stages and the way that we handle it causes some inefficiencies.  We have _almost_ the same exact modules for wpcalypso, horizon, staging, and production.  The only difference between them is the `config` module.  This single module difference essentially forces us to separate out bundles for each environment and call the bundler 4 times within a docker build. 

This PR seeks to move the `config` module to be sent by the server as part of the initial payload which:
1. cuts build times in a down because we don't need to rebuild docker image 4 times (and have 4 different versions of the bundler: `bundle-wpcalypso.js`, `bundle-production.js`, etc.
2. No longer need to generate a config before running unit tests


Speed tests on my machine:
- Building the docker image took 4:44.62 in this branch `/try/framework/ssr-config` 
- 11:44.25 on `master`

### To Test
- [ ] start calypso using make run
- [ ] run all the unit tests (and also circle ci)
- [ ] Open https://calypso.live/?branch=try/framework/ssr-config and make sure it still works
- [ ] build the docker image.  then run it in the 4 different environments and make sure it works as expected

related issues this should solve as well: 
- Generate config at runtime:  https://github.com/Automattic/wp-calypso/pull/10608
- Always use the right config during tests: https://github.com/Automattic/wp-calypso/issues/5856
